#Make configure to take into account EB's BLAS and LAPACK libraries 
#Use -parallel OMPCFLAG in case of intel icc compiler
#Use -parallel LDFLAGS in case of intel icc compiler
#Jan 24th 2017 by B. Hajgato (Free University Brussels - VUB)
--- configure.orig	2016-03-28 23:19:33.000000000 +0200
+++ configure	2017-01-24 13:52:36.353910986 +0100
@@ -1504,6 +1504,10 @@
 	    SHLIB_SUFFIX=".so"
 	    SHLIB_LD="${CC} -shared"
             LDFLAGS="-Wl,--export-dynamic"
+            if test ${CC} = "icc" ; then
+                    # Intel compiler
+                    LDFLAGS="-parallel -Wl,--export-dynamic"
+            fi
             LD_SEARCH_FLAGS='-Wl,-rpath-link,${LIB_RUNTIME_DIR}'
             LD_LIBRARY_PATH_VAR="LD_LIBRARY_PATH"
             ;;
@@ -10613,12 +10617,12 @@
 
 
 
-echo $ac_n "checking for dnrm2_ in -lblas""... $ac_c" 1>&6
-echo "configure:10618: checking for dnrm2_ in -lblas" >&5
+echo $ac_n "checking for dnrm2_ in $LIBBLAS""... $ac_c" 1>&6
+echo "configure:10618: checking for dnrm2_ in $LIBBLAS" >&5
 ac_lib_var=`echo blas'_'dnrm2_ | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-lblas $MATHLIB  $LIBS"
+LIBS="$LIBBLAS $MATHLIB  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 10624 "configure"
 #include "confdefs.h"
@@ -10645,16 +10649,16 @@
 
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  BLASLIB="$BLASLIB -lblas "
+  BLASLIB="$BLASLIB $LIBBLAS "
 else
   echo "$ac_t""no" 1>&6
 
-echo $ac_n "checking for dnrm2_ in -lblas""... $ac_c" 1>&6
-echo "configure:10654: checking for dnrm2_ in -lblas" >&5
+echo $ac_n "checking for dnrm2_ in $LIBBLAS""... $ac_c" 1>&6
+echo "configure:10654: checking for dnrm2_ in $LIBBLAS" >&5
 ac_lib_var=`echo blas'_'dnrm2_ | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-lblas $MATHLIB -lg2c $LIBS"
+LIBS="$LIBBLAS $MATHLIB -lg2c $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 10660 "configure"
 #include "confdefs.h"
@@ -10681,7 +10685,7 @@
 
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  BLASLIB="$BLASLIB -lblas -lg2c"
+  BLASLIB="$BLASLIB $LIBBLAS -lg2c"
 else
   echo "$ac_t""no" 1>&6
 
@@ -10864,12 +10868,12 @@
 
 # BLAS in PhiPACK libraries? (requires generic BLAS, too)
 if test $blas_ok = no; then
-	echo $ac_n "checking for sgemm_ in -lblas""... $ac_c" 1>&6
-echo "configure:10869: checking for sgemm_ in -lblas" >&5
+	echo $ac_n "checking for sgemm_ in $LIBBLAS""... $ac_c" 1>&6
+echo "configure:10869: checking for sgemm_ in $LIBBLAS" >&5
 ac_lib_var=`echo blas'_'sgemm_ | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-lblas  $LIBS"
+LIBS="$LIBBLAS  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 10875 "configure"
 #include "confdefs.h"
@@ -10901,7 +10905,7 @@
 ac_lib_var=`echo dgemm'_'dgemm_ | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-ldgemm -lblas $LIBS"
+LIBS="-ldgemm $LIBBLAS $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 10907 "configure"
 #include "confdefs.h"
@@ -10933,7 +10937,7 @@
 ac_lib_var=`echo sgemm'_'sgemm_ | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-lsgemm -lblas $LIBS"
+LIBS="-lsgemm $LIBBLAS $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 10939 "configure"
 #include "confdefs.h"
@@ -10960,7 +10964,7 @@
 
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  blas_ok=yes; BLASLIB="-lsgemm -ldgemm -lblas"
+  blas_ok=yes; BLASLIB="-lsgemm -ldgemm $LIBBLAS"
 else
   echo "$ac_t""no" 1>&6
 fi
@@ -11058,12 +11062,12 @@
 
 # Generic BLAS library
 if test $blas_ok = no; then
-	echo $ac_n "checking for sgemm_ in -lblas""... $ac_c" 1>&6
-echo "configure:11063: checking for sgemm_ in -lblas" >&5
+	echo $ac_n "checking for sgemm_ in $LIBBLAS""... $ac_c" 1>&6
+echo "configure:11063: checking for sgemm_ in $LIBBLAS" >&5
 ac_lib_var=`echo blas'_'sgemm_ | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-lblas  $LIBS"
+LIBS="$LIBBLAS  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 11069 "configure"
 #include "confdefs.h"
@@ -11090,7 +11094,7 @@
 
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  blas_ok=yes; BLASLIB="-lblas"
+  blas_ok=yes; BLASLIB="$LIBBLAS"
 else
   echo "$ac_t""no" 1>&6
 fi
@@ -11278,12 +11282,12 @@
 if test $lapack_ok = no; then
 	save_libs="$LIBS"; LIBS="$BLASLIB $MATHLIB $LIBS"
 	save_LDFLAGS="$LDFLAGS"; LDFLAGS="$LAPACKLIB $LDFLAGS"
-	echo $ac_n "checking for desgv_ in -llapack""... $ac_c" 1>&6
-echo "configure:11283: checking for desgv_ in -llapack" >&5
+	echo $ac_n "checking for desgv_ in $LIBLAPACK""... $ac_c" 1>&6
+echo "configure:11283: checking for desgv_ in $LIBLAPACK" >&5
 ac_lib_var=`echo lapack'_'desgv_ | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-llapack $FLIBS $LIBS"
+LIBS="$LIBLAPACK $FLIBS $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 11289 "configure"
 #include "confdefs.h"
@@ -11310,7 +11314,7 @@
 
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  lapack_ok=yes; LAPACKLIB="-llapack"
+  lapack_ok=yes; LAPACKLIB="$LIBLAPACK"
 else
   echo "$ac_t""no" 1>&6
 fi
@@ -11327,12 +11331,12 @@
 
 
 
-echo $ac_n "checking for dgesv_ in -llapack""... $ac_c" 1>&6
-echo "configure:11332: checking for dgesv_ in -llapack" >&5
+echo $ac_n "checking for dgesv_ in $LIBLAPACK""... $ac_c" 1>&6
+echo "configure:11332: checking for dgesv_ in $LIBLAPACK" >&5
 ac_lib_var=`echo lapack'_'dgesv_ | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-llapack $BLASLIB $MATHLIB  $LIBS"
+LIBS="$LIBLAPACK $BLASLIB $MATHLIB  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 11338 "configure"
 #include "confdefs.h"
@@ -11359,16 +11363,16 @@
 
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  LAPACKLIB="$LAPACKLIB -llapack "
+  LAPACKLIB="$LAPACKLIB $LIBLAPACK "
 else
   echo "$ac_t""no" 1>&6
 
-echo $ac_n "checking for dgesv_ in -llapack""... $ac_c" 1>&6
-echo "configure:11368: checking for dgesv_ in -llapack" >&5
+echo $ac_n "checking for dgesv_ in $LIBLAPACK""... $ac_c" 1>&6
+echo "configure:11368: checking for dgesv_ in $LIBLAPACK" >&5
 ac_lib_var=`echo lapack'_'dgesv_ | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-llapack $BLASLIB $MATHLIB -lg2c $LIBS"
+LIBS="$LIBLAPACK $BLASLIB $MATHLIB -lg2c $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 11374 "configure"
 #include "confdefs.h"
@@ -11395,7 +11399,7 @@
 
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  LAPACKLIB="$LAPACKLIB -llapack -lg2c"
+  LAPACKLIB="$LAPACKLIB $LIBLAPACK -lg2c"
 else
   echo "$ac_t""no" 1>&6
 
@@ -12798,9 +12802,10 @@
 if test x$GCC = xyes ; then
 	# GNU C compiler (>= 4.2.1)
 	OMPCFLAGS=-fopenmp
-elif test ${CC} = "icc" ; then
+fi
+if test ${CC} = "icc" ; then
 	# Intel compiler
-	OMPCFLAGS=-openmp
+	OMPCFLAGS=-parallel
 elif test ${CC} = "opencc" ; then
 	# AMD's x86 Open64 compiler
 	OMPCFLAGS=-mp
